File: summary/elixir/8_least_common_multiple.exs
defmodule MathUtils do
  def gcd(a, 0), do: a
  def gcd(a, b), do: gcd(b, rem(a, b))

  # 最小公倍数を計算する関数
  def lcm(a, b) when a == 0 or b == 0, do: 0
  def lcm(a, b), do: abs(a * b) / gcd(a, b)
end

# テスト
IO.puts(MathUtils.lcm(12, 18))  # => 36
IO.puts(MathUtils.lcm(5, 7))     # => 35

File: summary/dart/8_least_common_multiple.dart
int lcm(int a, int b) {
  if (a == 0 || b == 0) return 0;
  return (a * b).abs() ~/ gcd(a, b);
}

int gcd(int a, int b) {
  while (b != 0) {
    int temp = b;
    b = a % b;
    a = temp;
  }
  return a;
}

void main() {
  print(lcm(12, 18)); // 36
  print(lcm(5, 7)); // 35
}
